tests/foreman/sys/test_hooks.py

```diff
@@ -0,0 +1,159 @@
+
+
+@run_in_one_thread
+class ForemanHooksTestCase(TestCase):
+    """Test class for testing foreman hooks"""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create logger script to be executed via hooks"""
+        super(ForemanHooksTestCase, cls).setUpClass()
+        cls.org = 1  # using Default_Organization
+        cls.script_path = "{}/logger.sh".format(HOOKS_DIR)
+        ssh.command(
+            '''printf '#!/bin/sh\necho "$(date): Executed $1 hook'''
+            + ''' on object $2" > {0}' > {1}'''.format(
+                LOGS_DIR, cls.script_path)
+        )
+        ssh.command("chmod 774 {}".format(cls.script_path))
+        ssh.command("chown foreman:foreman {}".format(cls.script_path))
+        ssh.command("restorecon -RvF {}".format(HOOKS_DIR))
+
+    @classmethod
+    def tierDownClass(cls):
+        """clean up hook files"""
+        super(ForemanHooksTestCase, cls).tierDownClass()
+        ssh.command("rm -rf {}/*".format(HOOKS_DIR))
+
+    @tier3
+    def test_positive_host_hooks(self):
+        """Create hooks to be executed on host create, update and destroy
+
+        :id: 4fe35fda-1524-44f7-9221-96d1aeafc75c
+
+        :Steps:
+            1. Create hook directories with symlinks to logger script
+            for subset of supported events: create, update, destroy
+            2. Perform trigger actions (create, update and delete host)
+            3. Observe custom logs created by hook script
+
+        :expectedresults: hook scripts are executed at proper time
+        """
+        host_name = valid_hosts_list()[0]
+        expected_msg = "Executed {0} hook on object {1}"
+
+        # create hook destination directories with logger script
+        create_event = "create"
+        update_event = "update"
+        destroy_event = "destroy"
+        for event in [create_event, destroy_event, update_event]:
+            hook_dir = "{0}/host/managed/{1}".format(HOOKS_DIR, event)
+            ssh.command('''mkdir -p {}'''.format(hook_dir))
+            ssh.command('''ln -sf {0} {1}/'''.format(self.script_path, hook_dir))
+        result = ssh.command("systemctl restart httpd")
+        self.assertEqual(result.return_code, 0)
+
+        # delete host, check logs for hook activity
+        host = entities.Host(name=host_name).create()
+        self.assertEqual(host.name, '{0}.{1}'.format(host_name, host.domain.read().name))
+        result = ssh.command('cat {}'.format(LOGS_DIR))
+        self.assertEqual(result.return_code, 0)
+        self.assertIn(expected_msg.format(create_event, host_name), result.stdout[0])
+
+        # update host, check logs for hook activity
+        new_ip = gen_ipaddr()
+        host.ip = new_ip
+        host = host.update(['ip'])
+        self.assertEqual(host.ip, new_ip)
+        result = ssh.command('cat {}'.format(LOGS_DIR))
+        self.assertEqual(result.return_code, 0)
+        self.assertIn(expected_msg.format(update_event, host_name), result.stdout[0])
+
```

Comment:
rather repeating same steps, can be placed in another function e.g.
``` python
def _check_event_in_logfile(self, event_name, host_name):
  result = ssh.command('cat {}'.format(self.LOGS_DIR))
  self.assertEqual(result.return_code, 0)
  self.assertIn(expected_msg.format(event_name, host_name), result.stdout[0])
```
What do you think ?

Reply:
I see what you mean, will update